### ISSUE TITLE
Issue placing stop orders

### ISSUE THREAD

**Issue by axelarati** [dougdellolio/coinbasepro-csharp#223]
When I place a stop order I get the error "Invalid order_type stop"

I believe I found the error in the PlaceStopOrderAsync() function here: https://github.com/dougdellolio/coinbasepro-csharp/blob/master/CoinbasePro/Services/Orders/OrdersService.cs

According to the documentation https://docs.pro.coinbase.com/#place-a-new-order, shouldn't the OrderType be set to Market and the 'Stop' be set to StopType.Entry or StopType.Loss?

**Comment by dougdellolio** [comment]
@axelarati thanks for reporting. Please take a look at https://github.com/dougdellolio/coinbasepro-csharp/pull/224 and let me know what you think. I believe the current "PlaceStopLimitOrderAsync" isn't fully correct so I've removed it and re-worked "PlaceStopOrderAsync".  I've compared requests from placing an order through their UI to this library and they should match now.

**Comment by axelarati** [comment]
@dougdellolio Is it possible to place a stop (market) order with these new changes? Generally, I'm using stop orders to stop out of positions immediately, but I use stop limit orders as a way to take profits, i.e. once I hit a target price, I want to put a limit order at that target price (my reasons for not just putting a regular limit order are complicated).

Also, I believe your implementations of the StopType.Loss and StopType.Entry variables don't handle all cases. From my understanding, StopType.Loss triggers the stop order when the market hits the stop price on a down-tick, while StopType.Entry is triggered when the market hit the stop price on an up-tick. Your implementation sets the StopType to Entry for Buy orders and sets it to Loss for Sell orders. The problem with this is that when I want to use a stop limit order to take profit (as I mentioned above), since I am placing a Sell order, the function PlaceStopLimitOrderAsync() sets the StopType to Loss, which means my stop limit order only gets triggered if the market goes down after it has passed the stop price.
https://docs.pro.coinbase.com/#place-a-new-order
https://github.com/dougdellolio/coinbasepro-csharp/blob/master/CoinbasePro/Services/Orders/OrdersService.cs

I've looked into the KuCoin API, which is a different crypto broker, and it has this implemented as I described above. I also ran some tests and confirmed that the stop orders in KuCoin behave the way I expected.
https://docs.kucoin.com/#place-a-new-order

I think the only change you would need to make is to add the StopType as parameter to PlaceStopLimitOrderAsync().

**Comment by axelarati** [comment]
@dougdellolio I can work with that. I just wasn't sure what kind of stop orders I could place since the documentation didn't match what was available in the API. The PR looks great,  I don't see anything else that needs to be added for my purposes.

Thank you for all your help and very quick replies. It helps a lot with the development process, so  I really appreciate it.

**Comment by axelarati** [comment]
@dougdellolio I noticed a few problems once I tried working with the stop orders I get from Coinbase:
- No stop price shows up (only the limit price). It seems the documentation doesn't specify that there should be a stop price either when retrieving orders, but that is weird...
- When I request 'open' orders, the stop orders do not show up. They show up when I request 'active' orders, but 'active' orders aren't considered active for more than a few seconds. They also do not show up in the order web socket stream in the callback function OnOpenReceived()
- The order type on stop orders is 'limit', so I can't identify them (if they had a stop price, I would be able to). But I believe you intended for the order type to be 'Stop'.

**Comment by dougdellolio** [comment]
@axelarati 

Here's what I just tried on my end:

My code (not using async):

![image](https://user-images.githubusercontent.com/4141717/78844327-6fdc7280-79d3-11ea-9ffa-5d00ae134ca0.png)

This is the result on my sandbox:
![image](https://user-images.githubusercontent.com/4141717/78844107-bf6e6e80-79d2-11ea-9b6d-ea6835ed28cd.png)

openOrders returns to me 1 order (the one above)

The order type "Stop" doesn't seem to be supported anymore (I believe it was at some point). When stop price is provided the order type has to be Limit as Market isn't valid. You may be seeing no open orders if the stop price you specified was already met (number 5 from my above comment). Their documentation is definitely confusing and not complete around this order type.

If you feel comfortable, feel free to post some code and I can try to help clear up if there's any 
 more confusion.

**Comment by axelarati** [comment]
I think I figured out the problem. When I place a sell stop order with a stop price below the current price, it shows up as 'Active' in the API and the online sandbox UI. When I place an order with a stop price above the current price,  it shows up as 'Open'.

This leads me to believe that 'Active' means that the order is acknowledged by coinbase, but it isn't open as a limit or market order yet since the stop price hasn't been reached. I also confirmed this by looking at the Websocket Feed documentation. The function OnActivateReceived() is called for stop orders. https://docs.pro.coinbase.com/#the-full-channel

I'm assuming there is no way to query both 'Open' and 'Active' orders at the same time and I'll have to get all orders, then filter them. 

And the changes you linked in the your last reply should solve the problem of determining stop price. Thanks!
